### DISCUSSION TITLE
Depth Data for Nautical Charts

### DISCUSSION THREAD

**Discussion by quantenschaum** [osmandapp/OsmAnd#18116] [Maps Related]
Just for the record (resulting from https://github.com/osmandapp/OsmAnd/issues/18092), I managed to create an OBF containing depth contours and spot soundings from data leeched from https://www.geoseaportal.de/mapapps/resources/apps/navigation (currently down :disappointed:). The workflow is roughly like this:

- download data as GeoJSON
- import into QGIS
- filter the relevant rows
- add columns
  - `depth=VALDCO`
  - `contour=depth`
  - `contourtype=[5m,10m,20m,50m,100m]` with formula using `depth%type==0`
- export layer as geopackage only containing the 3 added columns
- import into JOSM via OpenData plugin
- save as OSM xml
- convert OSM to OBF using OsmAndMapCreator

Terrible, no command line tools, but it works.

OsmAnd does not display contours without a `contourtype` tag or values of `contourtype<5m`, so a 2m line will not show up on the map. I tweaked the [nautical map style](https://github.com/quantenschaum/mapping/raw/master/marine.render.xml) and the renderer [settings for contour lines](https://github.com/quantenschaum/mapping/raw/master/depthcontourlines.addon.render.xml). Use these, and the map will look like this.

![Screenshot_20230907-131703](https://github.com/osmandapp/OsmAnd/assets/3985599/6cc69478-f604-488c-9763-eb6fe7c3cbbe)

**Comment by miramikes** [comment]
Looks very good !

**Comment by quantenschaum** [comment]
## Data Sources

IMHO the depth data supplied by OsmAnd is just a proof of concept showcase, they are very inaccurate and unreliable and not at all usable for actual navigation or trip planning.

### Germany

The BSH provides access to nautical data in the [GeoSeaPortal](https://www.bsh.de/EN/DATA/GeoSeaPortal/geoseaportal_node.html). There is no documentation, but simply by examining what the webviewer does reveals some [WMS](https://en.wikipedia.org/wiki/Web_Map_Service) endpoints. Depth contours and areas can be found in

- https://www.geoseaportal.de/wss/service/NAUTHIS_Hydrography/guest
- https://www.geoseaportal.de/wss/service/NAUTHIS_SkinOfTheEarth/guest

 What is particularly nice is, that the data can be downloaded as `application/json;type=geojson`, so we can pull vector data from this server! You can download the entire dataset in one go.

BTW: They also provide a [tidal current atlas](https://en.wikipedia.org/wiki/Web_Map_Service) for free! In the webviewer you cannot select the time before/after HW, but if you use [QGIS](https://www.qgis.org/) as WMS client, you can explore the datasets and switch between different times.

- https://www.geoseaportal.de/wss/service/Gezeitenstrom_Nordsee/guest
- https://www.geoseaportal.de/wss/service/Gezeitenstrom_Kueste/guest

### Netherlands

RWS provides a full [ENC](https://en.wikipedia.org/wiki/Electronic_navigational_chart) with contours and spot soundings for free for some regions of NL like the Waddenzee and Zeeland.

- https://www.vaarweginformatie.nl/frp/main/#/page/infra_enc

Using [GDAL](https://gdal.org/programs/ogr2ogr.html) the vector data can be extracted from the ENC.

**Reply by singlerider** [reply]
I wasn't able to find a route to actually download the OSMAnd Nautical Contours data directly from the source code files.

Is the data they supply available at a known route?

My intention is to parse and post-process it for a FOSS game project of mine, assuming the license allows for it. I've looked in other discussions, but this is the closest one I've found so far that may have people that have figured something like this out.

**Reply by singlerider** [reply]
Funnily, I found this in the `LICENSES` file right after asking:

https://www.gebco.net/data_and_products/gridded_bathymetry_data/

I can probably parse this for my purposes 👍

**Comment by quantenschaum** [comment]
## `depth` Tag

Some objects like rocks or wrecks are tagged like `depth=2.3`. But this information is not displayed on the map, even though the data is compiled into the OBF as text already. It can be enabled by adding

```xml
    <case minzoom="10" tag="seamark:type" value="rock" textColor="$depthTextColor" textSize="14" textOrder="128" intersectionMargin="10" textDy="-12"/>
    <case minzoom="10" tag="seamark:type" value="wreck" textColor="$depthTextColor" textSize="14" textOrder="128" intersectionMargin="10" textDy="-10"/>
```

to the `<text>` section in the `render.xml`. The shield of the rock symbol (circle in background) clutters the map, it can be removed by setting `shield="null"`.

The result looks like this, just using the OOTB OFBs provided by OsmAnd.

![image](https://github.com/osmandapp/OsmAnd/assets/3985599/582f9c5d-266b-4749-858b-655b8fbeae53)

**Comment by quantenschaum** [comment]
# Spot Soundings

are already supported by the [nautical maps plugin](https://osmand.net/docs/user/plugins/nautical-charts). Just enable the "nautical depth" setting in the map config. Numbers will show up on the map for points tagged with

- `point=depth`
- `depth=<value>`

IMHO the font size is too small and should be increased/user configurable.

**Comment by quantenschaum** [comment]
## Contour Lines

The contours currently implemented in OsmAnd seem to be a copy of the height contours with some small changes. But depth contours in nav charts are different.

- Navigators are interested in the shallow areas, <10m and need high resolution there. Contours in deep water are nice to have, but no necessary for safe navigation, they might be interesting for fisherman or divers, but high resolution is not required in these areas. Depth contours are like log-scale.
- There are negative depths for drying heights in tidal waters. 
- Height and Depth have different [reference datum](https://en.wikipedia.org/wiki/Chart_datum). Depth usually refer to LAT in tidal waters and mean water level in non-tidal waters.
- Depending on the draft of your vessel, one particular depth contour is your "safety contour"

The [INT1](https://www.nauticalcharts.noaa.gov/publications/docs/us-chart-1/ChartNo1.pdf) section I contains explanations.

![image](https://github.com/osmandapp/OsmAnd/assets/3985599/88a4e9e7-3e52-404c-9b19-1311a14ebe05)

Different coloring schemes for the contours are used.

![image](https://github.com/osmandapp/OsmAnd/assets/3985599/29060381-2c9f-4999-9641-245b787dc1f5)
![image](https://github.com/osmandapp/OsmAnd/assets/3985599/250af257-5d2b-486f-968c-cdb212a63d55)


I would propose to only include integer depth levels (meters here, other units are a different thing) and tag the values (if available) <=5, 10, 20, 50, 100 with `contourtype=<depth>m`. This way you can

- hide/show lines at different zoom levels
- highlight an individual line

Other values may be in the dataset, but are tagged differently.

Currently OsmAnd does not show contours without `contourytype` tag and no lines <5m.

Tags for contourlines:

- `contour=depth` 
- `depth=<value>`
- `contourtype=<value>m` or `<interval>i` (optional, needed because depth is consumed as text during OBF creation)

This requires not only changing the `render.xml` but the `rendering_types.xml` used in the OBF creation also needs to be changed.

**Comment by quantenschaum** [comment]
Putting it all together, it looks like this.

![depth](https://github.com/osmandapp/OsmAnd/assets/3985599/4fcb07a5-9934-47cd-8e11-fbbf99ac1ec6)

You want this, too? Have a look at https://github.com/quantenschaum/mapping/blob/master/DEPTH.md

**Reply by miramikes** [reply]
Awesome ! I really hope that your work will be adopted by OsmAnd.
Meanwhile are you planning prepair depth files also for other countries?
https://github.com/quantenschaum/mapping/releases/tag/v1

**Reply by quantenschaum** [reply]
Thanks! I hope it is useful to others. I could prepare files for other areas, if I found data. Do you have any know data sources? The [US do provide ENCs for free](https://charts.noaa.gov/ENCs/ENCs.shtml). I could look into them.

**Reply by quantenschaum** [reply]
Here is data for the US. It Works, but needs quite some preprocessing in QGIS to filter duplicate data in overlapping regions and assigning the right tags. https://github.com/quantenschaum/mapping/releases/tag/v2

**Reply by miramikes** [reply]
Excellent... thank you for your work !!!

**Reply by miramikes** [reply]
Hello,
As I understand the hard part for more depth files is to find a good data source.
Maybe I found some for Norway.
https://kartkatalog.geonorge.no/metadata/depth-curves/871960a1-0f01-4c47-8f79-5d338b65197e

There is a note ...
Use limitations: Not approved for navigation.

But maybe it's worth to try ...
According to this data might be even 5x5m ...
https://www.mareano.no/en/maps-and-data/marine-geospatial-data

Mira

**Reply by quantenschaum** [reply]
Nice, but it is very coarse. Most shallow contour is at 20m. But I found another dataset on this server, the depth data used in nautical charts.
https://kartkatalog.geonorge.no/metadata/sjoekart-dybdedata/2751aacf-5472-4850-a208-3532a51c529a

**Reply by quantenschaum** [reply]
I created an OBF for Agder, as proof of concept, and added it to the release. The dataset is quite big.

**Reply by miramikes** [reply]
Excelent... thank you 👍

**Comment by miramikes** [comment]
Your release of lightsectors V5 is amazing... the obf file is worldwide or only for some countries?

**Reply by quantenschaum** [reply]
It's central/northern europe, but you can use script to create your own. Do you need any specific area?

**Reply by miramikes** [reply]
Areas which are important for me are Norway Sweden Denmark Germany Greek Italy Croatia Slovenia Korsika and Sardinia ... maybe GB
Basically whole Europe would be great 👍

**Reply by quantenschaum** [reply]
Well, the northern part of your list is mainly covered (except for the very north).

![image](https://github.com/osmandapp/OsmAnd/assets/3985599/9a42c18c-540d-41e9-b75d-4967a0a97024)

I can do more and add the files...

**Reply by quantenschaum** [reply]
Have a look again https://github.com/quantenschaum/mapping/releases/

**Reply by miramikes** [reply]
Oh 😮 ... many thanks !!!

**Reply by andrejmat** [reply]
I upload depths data but nothing change, still are wrong in my country (Croatia), wrong by 20-50m on some places. How to solve this? Thanks

**Reply by quantenschaum** [reply]
I didn't supply any depth data for Croatia.

**Reply by andrejmat** [reply]
Oh, i read previus post that user "miramikes" ask about that region and i was thinking that is done also, sorry. 
I have paper nautucal maps and totaly different depths but match with navionics, dont know from wich database are taken for OsmAnd, thanks for answer.

**Reply by quantenschaum** [reply]
There are depth data available in OsmAnd for download, but it is totally unclear where they come from and how old and accurate they are. I found them to be very inaccurate, too. They are effectively unusable.

**Comment by quantenschaum** [comment]
currently there files for download here http://waddenzee.duckdns.org/download/ updated regularly via cronjob

**Comment by quantenschaum** [comment]
I would like to propose here

1. update the depth data available (for germany I could provide up to date data)
2. change the tagging scheme of depth contours to include smaller values (<5m, these are the important ones)
3. add a tagging scheme for depth contour *areas*, have a look at http://waddenzee.duckdns.org/download/depth/
4. change the rendering of contours and areas 
 
All of that is already implemented as part of the proposal https://github.com/osmandapp/OsmAnd/issues/16894#issuecomment-2099952968

**Reply by josail** [reply]
@quantenschaum 
I saw some detailed nautical map incl. bathemetry on the east coast of UK close to Hull in one of the online exchanges that I had. If I remember correctly, the information was that data originated from Emodnet. If I search the Emodnet map data viewer, I find that not the whole of UK, but the whole of North Sea including UK coast has "detailed bathemetry data" https://emodnet.ec.europa.eu/geoviewer/#! and can be displayed in the map viewer. So I think this was the source of information. 

Emodnet is a European project and European Commission follows Open Accesss strategies and requirements for many of their projects. Thus, there should be a good chance to at least get this UK East coast and complete North Sea data. Find the Emodnet Map Viewer display as attached file.

**Comment by josail** [comment]
![Emodnet_Detailed_Bathemetry_NorthSea](https://github.com/osmandapp/OsmAnd/assets/93162192/66d7f7c6-e28e-4b18-87d4-43c1e88eaa30)

**Comment by josail** [comment]
![Emodnet_Detailed_Bathemetry_NorthSea_DataProduct](https://github.com/osmandapp/OsmAnd/assets/93162192/909a1356-b6f4-493f-8674-e91eb55de2c6)
See data product with green boundary. Does this lead to appropriate, open access usable data in Emodnet?

**Reply by josail** [reply]
maybe these links help opening this door?
https://ows.emodnet-bathymetry.eu/
https://emodnet.ec.europa.eu/en/emodnet-web-service-documentation
https://github.com/EMODnet

**Reply by quantenschaum** [reply]
Thanks. Didn't know that before. I will look into it.

**Reply by quantenschaum** [reply]
Some datasets are available from https://sextant.ifremer.fr/ like  
https://sextant.ifremer.fr/record/SDN_CPRD_1850_UAEms_2015/  
Pixel size = 0.00195313  Arc minute = 3.6m

**Reply by quantenschaum** [reply]
The entire dataset is here https://emodnet.ec.europa.eu/geonetwork/srv/eng/catalog.search#/metadata/ff3aff8a-cff1-44a3-a2c8-1910bf109f85  
it can be downloaded from https://erddap.emodnet.eu/erddap/files/bathymetry_2022/ 136GB

>Areas not covered by observations are completed by integrating GEBCO 2022 and IBCAO V4

So, it should be possible to extract contours and spot soundings from it.
